src/ssl.c

```diff
@@ -31852,57 +31944,72 @@ int wolfSSL_RAND_bytes(unsigned char* buf, int num)
 
     WOLFSSL_ENTER("wolfSSL_RAND_bytes");
 
+    if (initRandMethMutex != 0 && wc_LockMutex(&rand_methods_mutex) != 0) {
```

Comment:
I feel like this locking flow can be cleaned up.  Also why are none of the above also checking `initRandMethMutex`?

Reply:
This method is being called from our wolfCrypt tests, but wolfSSL_init is not called so this mutex was being used uninitialized and causing an error.  It wouldn't hurt to add it to the other functions, I'll add it and clean this up.